app/assets/javascripts/oxalis/model/accessors/flycam_accessor.js

```diff
@@ -9,14 +9,48 @@ import type { Matrix4x4 } from "libs/mjs";
 import { M4x4 } from "libs/mjs";
 import * as THREE from "three";
 import { getMaxZoomStep } from "oxalis/model/accessors/dataset_accessor";
+import { calculateUnzoomedBucketCount } from "oxalis/model/bucket_data_handling/bucket_picker_strategies/orthogonal_bucket_picker";
+import memoizeOne from "memoize-one";
+import { extraBucketsPerDim } from "oxalis/model/bucket_data_handling/bucket_picker_strategies/orthogonal_bucket_picker_constants";
 
 // All methods in this file should use constants.PLANE_WIDTH instead of constants.VIEWPORT_WIDTH
 // as the area that is rendered is only of size PLANE_WIDTH.
 // If VIEWPORT_WIDTH, which is a little bigger, is used instead, we end up with a data texture
 // that is shrinked a little bit, which leads to the texture not being in sync with the THREEjs scene.
 
-// maximum difference between requested coordinate and actual texture position
-const MAX_ZOOM_STEP_DIFF = constants.MAX_RENDERING_TARGET_WIDTH / constants.PLANE_WIDTH;
+// This function returns a value which indicates how much larger the rendered
+// plane can be than its original size **without** having to use the next
+// magnification. E.g., a value of two indicates that the viewport
+// can be 2 * viewport_width pixel wide while still being in zoom step 0.
+function unmemoizedCalculateMaxZoomStepDiff(dataSetScale: Vector3): number {
+  let maxZoomStep = 1;
+
+  // todo derive this from the textureBucketManager.maximumCapacity
+  const maximumCapacity = constants.MINIMUM_REQUIRED_BUCKET_CAPACITY;
+  while (
+    calculateUnzoomedBucketCount(dataSetScale, maxZoomStep) < maximumCapacity &&
+    maxZoomStep < 20
```

Comment:
I extracted the number into a constant and added a comment. This a really narrow edge case, which also doesn't do any harm even if we have datasets with more than 20 magnifications. Here's the comment:

```
  // This is more of a theoretical limit to avoid an endless loop, in case
  // the following while loop causes havoc for some reason. It means,
  // that even with the best GPU specs and weirdest dataset properties,
  // wk will at most render magnification 20 when being in zoom level 1.
```